Caught exception from MineColonies

The latest update (few minutes ago) crashes the server:

```
net.minecraftforge.fml.common.LoaderExceptionModCrash: Caught exception from MineColonies (minecolonies)
Caused by: java.lang.NoSuchMethodError: net.minecraft.util.math.MathHelper.clamp_int(III)I
	at com.minecolonies.coremod.configuration.ConfigurationHandler.getClampedInt(ConfigurationHandler.java:145)
	at com.minecolonies.coremod.configuration.ConfigurationHandler.init(ConfigurationHandler.java:52)
	at com.minecolonies.coremod.MineColonies.preInit(MineColonies.java:81)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at net.minecraftforge.fml.common.FMLModContainer.handleModStateEvent(FMLModContainer.java:616)
	at sun.reflect.GeneratedMethodAccessor9.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
	at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
	at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
	at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
	at com.google.common.eventbus.EventBus.post(EventBus.java:275)
	at net.minecraftforge.fml.common.LoadController.sendEventToModContainer(LoadController.java:243)
	at net.minecraftforge.fml.common.LoadController.propogateStateMessage(LoadController.java:221)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
	at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
	at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
	at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
	at com.google.common.eventbus.EventBus.post(EventBus.java:275)
	at net.minecraftforge.fml.common.LoadController.redirect$onPost$zzb000(LoadController.java:552)
	at net.minecraftforge.fml.common.LoadController.distributeStateMessage(LoadController.java:145)
	at net.minecraftforge.fml.common.Loader.preinitializeMods(Loader.java:624)
	at net.minecraftforge.fml.server.FMLServerHandler.beginServerLoading(FMLServerHandler.java:99)
	at net.minecraftforge.fml.common.FMLCommonHandler.onServerStart(FMLCommonHandler.java:328)
	at net.minecraft.server.dedicated.DedicatedServer.func_71197_b(DedicatedServer.java:121)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:431)
	at java.lang.Thread.run(Unknown Source)
net.minecraftforge.fml.common.LoaderExceptionModCrash: Caught exception from MineColonies (minecolonies)
Caused by: java.lang.NoSuchMethodError: net.minecraft.util.math.MathHelper.clamp_int(III)I
	at com.minecolonies.coremod.configuration.ConfigurationHandler.getClampedInt(ConfigurationHandler.java:145)
	at com.minecolonies.coremod.configuration.ConfigurationHandler.init(ConfigurationHandler.java:52)
	at com.minecolonies.coremod.MineColonies.preInit(MineColonies.java:81)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at net.minecraftforge.fml.common.FMLModContainer.handleModStateEvent(FMLModContainer.java:616)
	at sun.reflect.GeneratedMethodAccessor9.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
	at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
	at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
	at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
	at com.google.common.eventbus.EventBus.post(EventBus.java:275)
	at net.minecraftforge.fml.common.LoadController.sendEventToModContainer(LoadController.java:243)
	at net.minecraftforge.fml.common.LoadController.propogateStateMessage(LoadController.java:221)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
	at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
	at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
	at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
	at com.google.common.eventbus.EventBus.post(EventBus.java:275)
	at net.minecraftforge.fml.common.LoadController.redirect$onPost$zzb000(LoadController.java:552)
	at net.minecraftforge.fml.common.LoadController.distributeStateMessage(LoadController.java:145)
	at net.minecraftforge.fml.common.Loader.preinitializeMods(Loader.java:624)
	at net.minecraftforge.fml.server.FMLServerHandler.beginServerLoading(FMLServerHandler.java:99)
	at net.minecraftforge.fml.common.FMLCommonHandler.onServerStart(FMLCommonHandler.java:328)
	at net.minecraft.server.dedicated.DedicatedServer.func_71197_b(DedicatedServer.java:121)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:431)
	at java.lang.Thread.run(Unknown Source)
```

Do you have more information?

Minecraft 1.10?

2017-03-24 19:27 GMT-03:00 Tom <notifications@github.com>:

> The latest update (few minutes ago) crashes the server:
>
> net.minecraftforge.fml.common.LoaderExceptionModCrash: Caught exception from MineColonies (minecolonies)
> Caused by: java.lang.NoSuchMethodError: net.minecraft.util.math.MathHelper.clamp_int(III)I
> 	at com.minecolonies.coremod.configuration.ConfigurationHandler.getClampedInt(ConfigurationHandler.java:145)
> 	at com.minecolonies.coremod.configuration.ConfigurationHandler.init(ConfigurationHandler.java:52)
> 	at com.minecolonies.coremod.MineColonies.preInit(MineColonies.java:81)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
> 	at java.lang.reflect.Method.invoke(Unknown Source)
> 	at net.minecraftforge.fml.common.FMLModContainer.handleModStateEvent(FMLModContainer.java:616)
> 	at sun.reflect.GeneratedMethodAccessor9.invoke(Unknown Source)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
> 	at java.lang.reflect.Method.invoke(Unknown Source)
> 	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
> 	at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
> 	at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
> 	at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
> 	at com.google.common.eventbus.EventBus.post(EventBus.java:275)
> 	at net.minecraftforge.fml.common.LoadController.sendEventToModContainer(LoadController.java:243)
> 	at net.minecraftforge.fml.common.LoadController.propogateStateMessage(LoadController.java:221)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
> 	at java.lang.reflect.Method.invoke(Unknown Source)
> 	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
> 	at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
> 	at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
> 	at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
> 	at com.google.common.eventbus.EventBus.post(EventBus.java:275)
> 	at net.minecraftforge.fml.common.LoadController.redirect$onPost$zzb000(LoadController.java:552)
> 	at net.minecraftforge.fml.common.LoadController.distributeStateMessage(LoadController.java:145)
> 	at net.minecraftforge.fml.common.Loader.preinitializeMods(Loader.java:624)
> 	at net.minecraftforge.fml.server.FMLServerHandler.beginServerLoading(FMLServerHandler.java:99)
> 	at net.minecraftforge.fml.common.FMLCommonHandler.onServerStart(FMLCommonHandler.java:328)
> 	at net.minecraft.server.dedicated.DedicatedServer.func_71197_b(DedicatedServer.java:121)
> 	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:431)
> 	at java.lang.Thread.run(Unknown Source)
> net.minecraftforge.fml.common.LoaderExceptionModCrash: Caught exception from MineColonies (minecolonies)
> Caused by: java.lang.NoSuchMethodError: net.minecraft.util.math.MathHelper.clamp_int(III)I
> 	at com.minecolonies.coremod.configuration.ConfigurationHandler.getClampedInt(ConfigurationHandler.java:145)
> 	at com.minecolonies.coremod.configuration.ConfigurationHandler.init(ConfigurationHandler.java:52)
> 	at com.minecolonies.coremod.MineColonies.preInit(MineColonies.java:81)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
> 	at java.lang.reflect.Method.invoke(Unknown Source)
> 	at net.minecraftforge.fml.common.FMLModContainer.handleModStateEvent(FMLModContainer.java:616)
> 	at sun.reflect.GeneratedMethodAccessor9.invoke(Unknown Source)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
> 	at java.lang.reflect.Method.invoke(Unknown Source)
> 	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
> 	at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
> 	at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
> 	at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
> 	at com.google.common.eventbus.EventBus.post(EventBus.java:275)
> 	at net.minecraftforge.fml.common.LoadController.sendEventToModContainer(LoadController.java:243)
> 	at net.minecraftforge.fml.common.LoadController.propogateStateMessage(LoadController.java:221)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
> 	at java.lang.reflect.Method.invoke(Unknown Source)
> 	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
> 	at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
> 	at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
> 	at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
> 	at com.google.common.eventbus.EventBus.post(EventBus.java:275)
> 	at net.minecraftforge.fml.common.LoadController.redirect$onPost$zzb000(LoadController.java:552)
> 	at net.minecraftforge.fml.common.LoadController.distributeStateMessage(LoadController.java:145)
> 	at net.minecraftforge.fml.common.Loader.preinitializeMods(Loader.java:624)
> 	at net.minecraftforge.fml.server.FMLServerHandler.beginServerLoading(FMLServerHandler.java:99)
> 	at net.minecraftforge.fml.common.FMLCommonHandler.onServerStart(FMLCommonHandler.java:328)
> 	at net.minecraft.server.dedicated.DedicatedServer.func_71197_b(DedicatedServer.java:121)
> 	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:431)
> 	at java.lang.Thread.run(Unknown Source)
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Minecolonies/minecolonies/issues/735>, or mute the
> thread
> <https://github.com/notifications/unsubscribe-auth/AGI9y_5hHuURfAo-Oa9bEmuqoo37jGbkks5rpENtgaJpZM4Mo1Zg>
> .
>

Found it. On it.

2017-03-24 19:32 GMT-03:00 Ray Neiheiser <ray.neiheiser@gmail.com>:

> Do you have more information?
>
> Minecraft 1.10?
>
> 2017-03-24 19:27 GMT-03:00 Tom <notifications@github.com>:
>
>> The latest update (few minutes ago) crashes the server:
>>
>> net.minecraftforge.fml.common.LoaderExceptionModCrash: Caught exception from MineColonies (minecolonies)
>> Caused by: java.lang.NoSuchMethodError: net.minecraft.util.math.MathHelper.clamp_int(III)I
>> 	at com.minecolonies.coremod.configuration.ConfigurationHandler.getClampedInt(ConfigurationHandler.java:145)
>> 	at com.minecolonies.coremod.configuration.ConfigurationHandler.init(ConfigurationHandler.java:52)
>> 	at com.minecolonies.coremod.MineColonies.preInit(MineColonies.java:81)
>> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>> 	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
>> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
>> 	at java.lang.reflect.Method.invoke(Unknown Source)
>> 	at net.minecraftforge.fml.common.FMLModContainer.handleModStateEvent(FMLModContainer.java:616)
>> 	at sun.reflect.GeneratedMethodAccessor9.invoke(Unknown Source)
>> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
>> 	at java.lang.reflect.Method.invoke(Unknown Source)
>> 	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
>> 	at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
>> 	at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
>> 	at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
>> 	at com.google.common.eventbus.EventBus.post(EventBus.java:275)
>> 	at net.minecraftforge.fml.common.LoadController.sendEventToModContainer(LoadController.java:243)
>> 	at net.minecraftforge.fml.common.LoadController.propogateStateMessage(LoadController.java:221)
>> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>> 	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
>> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
>> 	at java.lang.reflect.Method.invoke(Unknown Source)
>> 	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
>> 	at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
>> 	at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
>> 	at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
>> 	at com.google.common.eventbus.EventBus.post(EventBus.java:275)
>> 	at net.minecraftforge.fml.common.LoadController.redirect$onPost$zzb000(LoadController.java:552)
>> 	at net.minecraftforge.fml.common.LoadController.distributeStateMessage(LoadController.java:145)
>> 	at net.minecraftforge.fml.common.Loader.preinitializeMods(Loader.java:624)
>> 	at net.minecraftforge.fml.server.FMLServerHandler.beginServerLoading(FMLServerHandler.java:99)
>> 	at net.minecraftforge.fml.common.FMLCommonHandler.onServerStart(FMLCommonHandler.java:328)
>> 	at net.minecraft.server.dedicated.DedicatedServer.func_71197_b(DedicatedServer.java:121)
>> 	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:431)
>> 	at java.lang.Thread.run(Unknown Source)
>> net.minecraftforge.fml.common.LoaderExceptionModCrash: Caught exception from MineColonies (minecolonies)
>> Caused by: java.lang.NoSuchMethodError: net.minecraft.util.math.MathHelper.clamp_int(III)I
>> 	at com.minecolonies.coremod.configuration.ConfigurationHandler.getClampedInt(ConfigurationHandler.java:145)
>> 	at com.minecolonies.coremod.configuration.ConfigurationHandler.init(ConfigurationHandler.java:52)
>> 	at com.minecolonies.coremod.MineColonies.preInit(MineColonies.java:81)
>> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>> 	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
>> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
>> 	at java.lang.reflect.Method.invoke(Unknown Source)
>> 	at net.minecraftforge.fml.common.FMLModContainer.handleModStateEvent(FMLModContainer.java:616)
>> 	at sun.reflect.GeneratedMethodAccessor9.invoke(Unknown Source)
>> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
>> 	at java.lang.reflect.Method.invoke(Unknown Source)
>> 	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
>> 	at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
>> 	at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
>> 	at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
>> 	at com.google.common.eventbus.EventBus.post(EventBus.java:275)
>> 	at net.minecraftforge.fml.common.LoadController.sendEventToModContainer(LoadController.java:243)
>> 	at net.minecraftforge.fml.common.LoadController.propogateStateMessage(LoadController.java:221)
>> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>> 	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
>> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
>> 	at java.lang.reflect.Method.invoke(Unknown Source)
>> 	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
>> 	at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
>> 	at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
>> 	at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
>> 	at com.google.common.eventbus.EventBus.post(EventBus.java:275)
>> 	at net.minecraftforge.fml.common.LoadController.redirect$onPost$zzb000(LoadController.java:552)
>> 	at net.minecraftforge.fml.common.LoadController.distributeStateMessage(LoadController.java:145)
>> 	at net.minecraftforge.fml.common.Loader.preinitializeMods(Loader.java:624)
>> 	at net.minecraftforge.fml.server.FMLServerHandler.beginServerLoading(FMLServerHandler.java:99)
>> 	at net.minecraftforge.fml.common.FMLCommonHandler.onServerStart(FMLCommonHandler.java:328)
>> 	at net.minecraft.server.dedicated.DedicatedServer.func_71197_b(DedicatedServer.java:121)
>> 	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:431)
>> 	at java.lang.Thread.run(Unknown Source)
>>
>> —
>> You are receiving this because you are subscribed to this thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Minecolonies/minecolonies/issues/735>, or mute the
>> thread
>> <https://github.com/notifications/unsubscribe-auth/AGI9y_5hHuURfAo-Oa9bEmuqoo37jGbkks5rpENtgaJpZM4Mo1Zg>
>> .
>>
>
>

--- a/src/main/java/com/minecolonies/coremod/colony/ColonyList.java
+++ b/src/main/java/com/minecolonies/coremod/colony/ColonyList.java
@@ -96,8 +96,8 @@ public final class ColonyList<T extends IColony> implements Iterable<T>
         }
 
         int emptyIds = colony.getID() - 1;
-        while(emptyIdsit
-                != 0  && list[emptyIds] == null)
+        while(emptyIds
+                > 0  && list[emptyIds] == null)
         {
             emptyIds--;
             nullIndices.add(emptyIds);

--- a/src/main/java/com/minecolonies/coremod/colony/ColonyList.java
+++ b/src/main/java/com/minecolonies/coremod/colony/ColonyList.java
@@ -96,7 +96,8 @@ public final class ColonyList<T extends IColony> implements Iterable<T>
         }
 
         int emptyIds = colony.getID() - 1;
-        while(list[emptyIds] == null)
+        while(emptyIdsit
+                != 0  && list[emptyIds] == null)
         {
             emptyIds--;
             nullIndices.add(emptyIds);

--- a/src/main/java/com/minecolonies/coremod/configuration/ConfigurationHandler.java
+++ b/src/main/java/com/minecolonies/coremod/configuration/ConfigurationHandler.java
@@ -1,7 +1,7 @@
 package com.minecolonies.coremod.configuration;
 
-import net.minecraft.util.math.MathHelper;
 import net.minecraftforge.common.config.Configuration;
+import net.minecraft.util.math.MathHelper;
 
 import java.io.File;
 

